api/src/main/java/org/pmiops/workbench/api/DiskController.java

```diff
@@ -0,0 +1,117 @@
+import org.pmiops.workbench.exceptions.NotFoundException;
+import org.pmiops.workbench.leonardo.model.LeonardoGetPersistentDiskResponse;
+import org.pmiops.workbench.leonardo.model.LeonardoListPersistentDiskResponse;
+import org.pmiops.workbench.leonardo.model.LeonardoRuntimeStatus;
+import org.pmiops.workbench.model.Disk;
+import org.pmiops.workbench.model.EmptyResponse;
+import org.pmiops.workbench.model.WorkspaceAccessLevel;
+import org.pmiops.workbench.notebooks.LeonardoNotebooksClient;
+import org.pmiops.workbench.utils.mappers.LeonardoMapper;
+import org.pmiops.workbench.workspaces.WorkspaceAuthService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class DiskController implements DiskApiDelegate {
+  private static final Logger log = Logger.getLogger(DiskController.class.getName());
+  private final LeonardoNotebooksClient leonardoNotebooksClient;
+  private final WorkspaceDao workspaceDao;
+  private final WorkspaceAuthService workspaceAuthService;
+  private final LeonardoMapper leonardoMapper;
+
+  @Autowired
+  public DiskController(
+      LeonardoNotebooksClient leonardoNotebooksClient,
+      WorkspaceAuthService workspaceAuthService,
+      WorkspaceDao workspaceDao,
+      LeonardoMapper leonardoMapper) {
+    this.leonardoNotebooksClient = leonardoNotebooksClient;
+    this.workspaceAuthService = workspaceAuthService;
+    this.workspaceDao = workspaceDao;
+    this.leonardoMapper = leonardoMapper;
+  }
+
+  private DbWorkspace lookupWorkspace(String workspaceNamespace)
+      throws org.pmiops.workbench.exceptions.NotFoundException {
+    return workspaceDao
+        .getByNamespace(workspaceNamespace)
+        .orElseThrow(() -> new NotFoundException("Workspace not found: " + workspaceNamespace));
+  }
+
+  @Override
+  public ResponseEntity<Disk> getDisk(String workspaceNamespace) {
+    DbWorkspace dbWorkspace = lookupWorkspace(workspaceNamespace);
+    String googleProject = dbWorkspace.getGoogleProject();
+    try {
+      LeonardoGetPersistentDiskResponse response;
+      List<LeonardoListPersistentDiskResponse> responseList =
+          leonardoNotebooksClient.listPersistentDiskByProject(googleProject, false);
+      responseList.sort(
+          new Comparator<LeonardoListPersistentDiskResponse>() {
+            @Override
+            public int compare(
+                LeonardoListPersistentDiskResponse t1, LeonardoListPersistentDiskResponse t2) {
+              return -1 * t1.getId().compareTo(t2.getId());
+            }
+          });
+      response =
+          !responseList.isEmpty()
+              ? leonardoNotebooksClient.getPersistentDisk(
+                  googleProject, responseList.get(0).getName())
+              : null;
+
+      if (response != null && LeonardoRuntimeStatus.ERROR.equals(response.getStatus())) {
```

Comment:
I am not sure how this part would work.
1: Is `LeonardoRuntimeStatus` supposed be used in PdStatus?  
2: What is the response if we saw error from Leo? Seems we just logging then return okay? 
It does seems to be right

Reply:
Fixed. Now we'll logging and return the disk with error status.